### DISCUSSION TITLE
why my array can't push_back in the json

### DISCUSSION THREAD

**Discussion by zxz970104** [nlohmann/json#3911] [Q&A]
my code is

> json inform = bundle_obj->get_state_obj()->get_inform();
cout  << inform << endl;
this->bundles_obj->lastTask->push_back(json::object_t::value_type("state", inform));
cout << "====" << this->bundles_obj->lastTask->at("state").get<json>() << endl;

my output is：

> [{"entity":"sys.鍥藉唴鍩庡競","extra":null,"intent":"瀵艰埅","name":"璧风偣鍚嶇О","poi_postfix":"","raw_pinyin":"su zhou","raw_value":"鑻忓窞","refer":false,"turn":11,"value":"鑻忓窞"},{"entity":"sys.鍥藉唴鍩庡競","extra":null,"intent":"瀵艰埅","name":"缁堢偣鍚嶇О","poi_postfix":"","raw_pinyin":"bei jing","raw_value":"鍖椾含","refer":false,"turn":11,"value":"鍖椾含"}]
====null

and at my test code：

> json js2 = json::parse(R"(
            {
                "pi": 3.141,
                "happy": true,
                "name": "Niels",
                "nothing": null,
                "answer": {
                    "everything": 42
                },
                "bool": true,
                "array": [{"x": 1, "y":2}, {"a": 3, "b":4}],
                "empty": {},
                "list": [1, 0, 2],
                "object": {
                    "currency": "USD",
                    "value": 42.99
                }
            }
        )");
 js2.push_back(json::object_t::value_type("zxz", js2.at("array")));
 cout << js2 << endl;

the output is：

> {"answer":{"everything":42},"array":[{"x":1,"y":2},{"a":3,"b":4}],"bool":true,"empty":{},"happy":true,"list":[1,0,2],"name":"Niels","nothing":null,"object":{"currency":"USD","value":42.99},"pi":3.141,**"zxz":[{"x":1,"y":2},{"a":3,"b":4}]**}

**Comment by nlohmann** [comment]
What do you expect?
